plugins/source/azure/Makefile

```diff
@@ -19,21 +9,34 @@ install-tools:
 install-hooks:
 	pre-commit install
 
-.PHONY: gen-docs
-gen-docs:
-	rm -rf ./docs/tables/*
-	go run main.go doc ./docs/tables
+# unit tests
+.PHONY: test
+test:
+	go test -timeout 3m ./...
 
 .PHONY: lint
 lint:
-	golangci-lint run --config ../../.golangci.yml 
+	golangci-lint run --config ../../.golangci.yml
 
 .PHONY: gen-code
 gen-code:
-	grep -rl '// Code generated by codegen; DO NOT EDIT.' resources/services/* | xargs rm
 	go install golang.org/x/tools/cmd/goimports@latest
+	grep -rl '// Code generated by codegen; DO NOT EDIT.' resources/services/* | xargs rm
 	go run codegen/main.go
 
-# All gen targets
+# Generate mocks for mock/unit testing
+.PHONY: generate-mocks
+generate-mocks:
+	rm -rf ./client/mocks/*
+	go generate ./client/services/...
+
+# Generate docs
+.PHONY: gen-docs
+gen-docs:
+	rm -rf ./docs/tables/*
+	go run main.go doc ./docs/tables
+
+
+# All main gen targets
 .PHONY: gen
 gen: gen-code gen-docs
```

Comment:
```suggestion
gen: gen-code gen-mocks gen-docs
```

Reply:
decided to have a separate target for clients regen + mocks. should it be called in the `gen` target?